src/test/test_verify.cpp

```diff
@@ -0,0 +1,369 @@
+    TEST_SECTION("linux", "tracex7_kern.o", "kprobe/open_ctree");
+    TEST_SECTION("linux", "xdp_adjust_tail_kern.o", "xdp_icmp");
+    TEST_SECTION("linux", "xdp_fwd_kern.o", "xdp_fwd");
+    TEST_SECTION("linux", "xdp_fwd_kern.o", "xdp_fwd_direct");
+    TEST_SECTION("linux", "xdp_monitor_kern.o", "tracepoint/xdp/xdp_cpumap_enqueue");
+    TEST_SECTION("linux", "xdp_monitor_kern.o", "tracepoint/xdp/xdp_cpumap_kthread");
+    TEST_SECTION("linux", "xdp_monitor_kern.o", "tracepoint/xdp/xdp_devmap_xmit");
+    TEST_SECTION("linux", "xdp_monitor_kern.o", "tracepoint/xdp/xdp_exception");
+    TEST_SECTION("linux", "xdp_monitor_kern.o", "tracepoint/xdp/xdp_redirect");
+    TEST_SECTION("linux", "xdp_monitor_kern.o", "tracepoint/xdp/xdp_redirect_err");
+    TEST_SECTION("linux", "xdp_monitor_kern.o", "tracepoint/xdp/xdp_redirect_map");
+    TEST_SECTION("linux", "xdp_monitor_kern.o", "tracepoint/xdp/xdp_redirect_map_err");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "xdp_cpu_map0");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "xdp_cpu_map1_touch_data");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "xdp_cpu_map2_round_robin");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "xdp_cpu_map3_proto_separate");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "xdp_cpu_map4_ddos_filter_pktgen");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "xdp_cpu_map5_lb_hash_ip_pairs");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "tracepoint/xdp/xdp_cpumap_enqueue");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "tracepoint/xdp/xdp_cpumap_kthread");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "tracepoint/xdp/xdp_exception");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "tracepoint/xdp/xdp_redirect_err");
+    TEST_SECTION("linux", "xdp_redirect_cpu_kern.o", "tracepoint/xdp/xdp_redirect_map_err");
+    TEST_SECTION("linux", "xdp_redirect_kern.o", "xdp_redirect");
+    TEST_SECTION("linux", "xdp_redirect_kern.o", "xdp_redirect_dummy");
+    TEST_SECTION("linux", "xdp_redirect_map_kern.o", "xdp_redirect_dummy");
+    TEST_SECTION("linux", "xdp_redirect_map_kern.o", "xdp_redirect_map");
+    TEST_SECTION("linux", "xdp_router_ipv4_kern.o", "xdp_router_ipv4");
+    TEST_SECTION("linux", "xdp_rxq_info_kern.o", "xdp_prog0");
+    TEST_SECTION("linux", "xdp_sample_pkts_kern.o", "xdp_sample");
+    TEST_SECTION("linux", "xdp_tx_iptunnel_kern.o", "xdp_tx_iptunnel");
+    TEST_SECTION("linux", "xdp1_kern.o", "xdp1");
+    TEST_SECTION("linux", "xdp2_kern.o", "xdp1");
+    TEST_SECTION("linux", "xdp2skb_meta_kern.o", "tc_mark");
+    TEST_SECTION("linux", "xdp2skb_meta_kern.o", "xdp_mark");
+    TEST_SECTION("linux", "xdpsock_kern.o", "xdp_sock");
+}
+
+// The following eBPF programs currently fail verification.
+// If the verifier is later updated to accept them, these should move
+// up to the previous test case.
+TEST_CASE("unverified linux samples", "[verify][linux][failure]") {
```

Comment:
We should return an "unsupported feature" in these cases (map in map).
Just a note; I am going to merge this as it is.